oioioi/contests/utils.py

```diff
@@ -247,6 +256,96 @@ def visible_rounds(request):
     return [r for r in queryset if controller.can_see_round(request, r)]
 
 
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
+
+
+def get_contest_dates(request):
+    """Returns the end_date of the latest round and the start_date
+    of the earliest round.
+    """ 
+    if isinstance(request, HttpRequest):
+        rtimes = rounds_times(request, request.contest)
+    else:
+        rtimes = generic_rounds_times(None, request.contest)
+
+    ends = [
+        rt.get_end()
+        for rt in rtimes.values()
+    ]
+    starts = [
+        rt.get_start()
+        for rt in rtimes.values()
+    ]
+
+    if (not starts or starts.count(None)):
+        min_start = -1
+    else:
+        min_start = min(starts)
+
+    if (not ends or ends.count(None)):
+        max_end = -1
+    else:
+        max_end = max(ends)
+
+    return min_start, max_end
+
+
+@request_cached
+def get_problems_sumbmission_limit(request):
+    """Returns the list of distinct submission limits in the current contest.
+    """
+    controller = request.contest.controller
+    queryset = (
+        ProblemInstance.objects.filter(contest=request.contest)
+        .prefetch_related('round')
+    )
+
+    if (queryset is None):
+        return None
+    
+    limits = set()
+    for p in queryset:
+        limits.add(controller.get_submissions_limit(request, p))
+
+    return sorted(limits)
```

Comment:
I think that for contests without problems the contest's default submissions limit should be returned here.

Reply:
Good idea, changed.